misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -101,18 +101,19 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
     }
   }
 
-  /** Set a ByteArray value if it doesn't already exist. */
-  override fun setnx(key: String, value: ByteString) {
-    jedisPool.resource.use { jedis ->
-      jedis.setnx(key.toByteArray(charset), value.toByteArray())
+  /** Set a ByteArray value if it doesn't already exist. Returns 1 if set and 0, otherwise */
+  override fun setnx(key: String, value: ByteString): Boolean {
+    return jedisPool.resource.use { jedis ->
+      jedis.setnx(key.toByteArray(charset), value.toByteArray()) == 1L
     }
   }
 
-  /** Set a ByteArray value if it doesn't already exist with an expiration. */
-  override fun setnx(key: String, expiryDuration: Duration, value: ByteString) {
-    val setParams = SetParams.setParams().ex(expiryDuration.seconds).nx()
-    jedisPool.resource.use { jedis ->
-      jedis.set(key.toByteArray(charset), value.toByteArray(), setParams)
+  /** Set a ByteArray value if it doesn't already exist with an expiration. Returns 1 if set and 0, otherwise  */
+  override fun setnx(key: String, expiryDuration: Duration, value: ByteString): Boolean {
+    return jedisPool.resource.use { jedis ->
+      jedis.setnx(key.toByteArray(charset), value.toByteArray())
+        .also { setResult -> if (setResult == 1L) jedis.expire(key, expiryDuration.seconds) } == 1L
```

Comment:
jedis doesn't have a setnx with expiry so I had to make 2 commands

Reply:
ugh, so annoying. But I like this better than what we had.